### ISSUE TITLE
Branch is leaking on iOS 9

### ISSUE THREAD

**Issue by romk1n** [BranchMetrics/iOS-Deferred-Deep-Linking-SDK#187]
Hi looks like there are few leaks detected by instruments from the library version 0.11.4

in `- (void)genericHTTPRequest:(NSURLRequest *)request log:(BOOL)log callback:(BNCServerCallback)callback `

here the screenshot from the instruments please have a look. Thanks!

![screen shot 2015-09-18 at 6 21 15 pm](https://cloud.githubusercontent.com/assets/1556000/9974546/9c3fbac2-5e48-11e5-9cf3-976c9676d41c.png)

**Comment by aaustin** [comment]
Thanks for the note @romk1n - we'll dig in and let you know after we know more.

**Comment by shortstuffsushi** [comment]
@romk1n, it would be useful if you could provide either the line number, or the message about what is leaking as well. I would check this myself, but I'm away from my actual laptop with Xcode at the moment.

**Comment by romk1n** [comment]
@shortstuffsushi sure! `BNCServerInterface` [here](https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/blob/master/Branch-SDK/Branch-SDK/BNCServerInterface.m#L78)

**Comment by shortstuffsushi** [comment]
@derrickstaten I believe your changes to use `NSURLSession` needs to include `[session finishTaskAndInvalidate]` to prevent this leak, per [this SO post](http://stackoverflow.com/questions/21554987/nsurlsession-memory-leaks-occur-when-using-web-services-in-ios) and the [corresponding Apple docs](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/occ/instm/NSURLSession/finishTasksAndInvalidate).

**Comment by derrickstaten** [comment]
Thanks, all. this has now been addressed with PR 188: https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/pull/188

This will be a part of release `0.11.5`

**Comment by romk1n** [comment]
@derrickstaten thanks a lot!

**Comment by romk1n** [comment]
@derrickstaten i just updated to 0.11.5 and still getting leaks in

 `- (void)genericHTTPRequest:(NSURLRequest *)request retryNumber:(NSInteger)retryNumber log:(BOOL)log callback:(BNCServerCallback)callback retryHandler:(NSURLRequest *(^)(NSInteger))retryHandler`

also in `- (void)af_resume`

**Comment by derrickstaten** [comment]
@romk1n we'll dig into this. I thought we had definitively fixed this. Sorry for getting your hopes up. Also, feel free to open a PR if you determine where the issue lies. 

We'll try to get this patched over the weekend or early next week.

**Comment by romk1n** [comment]
@derrickstaten not sure how, but when pod installs version 0.11.5 the changes made are not present in the code.

**Comment by derrickstaten** [comment]
@romk1n you're right. I'm able to reproduce -- that code is missing. I'm not familiar with our Cocoapods release process, so I'll likely have to reach out for help.

**Comment by shortstuffsushi** [comment]
This is due to the fact that there are both a branch and a tag named `0.11.5`. Deleting the branch will work, but @romk1n will likely need to delete his `Podfile.lock` and `Pods` directory to pick up the update.

**Comment by shortstuffsushi** [comment]
@romk1n but only after @derrickstaten has confirmed the branch deleted.

**Comment by romk1n** [comment]
@shortstuffsushi @derrickstaten sounds great, can you just ping me when it is done? thanks for being so prompt!

**Comment by derrickstaten** [comment]
@romk1n per instructions from @shortstuffsushi you'll also need to run `pod cache clean Branch` and remove 0.11.5. Then upon reinstall you'll get the latest.

**Comment by derrickstaten** [comment]
@romk1n there will also be a new release momentarily, so you can simply run `pod update`

**Comment by romk1n** [comment]
@derrickstaten thanks, what's bizarre is not i get 19 compile errors mainly because of the `NS_ASSUME_NONNULL_BEGIN`

**Comment by romk1n** [comment]
Looks good now! Many Thanks!
